.github/workflows/JirabotMerge.yml

```diff
@@ -0,0 +1,207 @@
+
+on:
+  push:
+    branches:
+      - "master"
+      - "candidate-*"
+
+jobs:
+  jirabot:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: "actions/setup-python@v2"
+        with:
+          python-version: "3.8"
+      - name: "Install dependencies"
+        run: |
+          set -xe
+          python -VV
+          python -m site
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --upgrade jira
+      - name: "Run"
+        env:
+          JIRABOT_USERNAME : ${{ secrets.JIRABOT_USERNAME }}
+          JIRABOT_PASSWORD : ${{ secrets.JIRABOT_PASSWORD }}
+          JIRA_URL : ${{ secrets.JIRA_URL }}
+          PULL_REQUEST_NUMBER : ${{ github.event.pull_request.number }}
+          PULL_REQUEST_TITLE : ${{ github.event.pull_request.title }}
+          PULL_REQUEST_AUTHOR_NAME : ${{ github.event.pull_request.user.login }}
+          PULL_URL: ${{ github.event.pull_request.html_url }}
+          COMMENTS_URL: ${{ github.event.pull_request.comments_url }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BRANCH_NAME: ${{ github.ref_name }}
+
+        run: |
+            import os
+            import re
+            import subprocess
+            import time
+            from jira.client import JIRA
+
+            def extractVersion(versionStr):
+                parts = versionStr.split('.')
+                if len(parts) != 3:
+                    print('Invalid version: ' + version)
+                    sys.exit(1)
+                if parts[2].lower() == 'x':
+                    parts[2] = '0'
+
+                major, minor, point = map(int, parts)
+                return [major, minor, point]
+
+            def getTagVersionForCmd(cmd):
+                versionPattern = re.compile(r".*([0-9]+\.[0-9]+\.[0-9]+).*")
+
+                # Get latest release version
+                gitTagProcess = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+                (output, err) = gitTagProcess.communicate()
+                gitTagProcessStatus = gitTagProcess.wait()
+
+                if gitTagProcessStatus != 0:
+                    print('Unable to retrieve latest git tag.')
+                    sys.exit(1)
+
+                latestGitTag = str(output)
+
+                versionMatch = versionPattern.match(latestGitTag)
+                if versionMatch:
+                    return extractVersion(versionMatch.group(1))
+                else:
+                    print('Unable to extract version from git tag.')
+                    sys.exit(2)
+
+            def buildVersionString(version):
+                major, minor, point = map(int, version)
+                return f"{major}.{minor}.{point}"
+
+            def generateFixVersionList(jira, branchName):
+
+                latestVersion = getTagVersionForCmd("git tag --list 'hpcc4j_*-release' --sort=-v:refname | head -n 1")
+
+                # If we are merging into master we assume it is going into the next minor release
```

Comment:
As discussed, there's a scenario where master is reserved for next major, I wonder if we can infer this situation if we find a branch for the next minor but no corresponding tags...
master version 10.1.0
latest tag 10.4.10
branch 10.6.x exists